frontend/pages/file-storage/FileList.test.jsx

```diff
@@ -0,0 +1,256 @@
+    expect(ascendingIcon).toBeInTheDocument();
+  });
+
+  it('shows the descending icon if currentSortOrder is desc', () => {
+    render(
+      <MemoryRouter>
+        <FileList {...mockProps} currentSortOrder="desc" />
+      </MemoryRouter>,
+    );
+    const sortButton = screen.getByLabelText('Sort by name');
+    const descendingIcon = within(sortButton).getByLabelText('descending sort icon');
+    expect(descendingIcon).toBeInTheDocument();
+  });
+
+  it('shows the unsorted icon on headers that are not currently sorted', () => {
+    render(
+      <MemoryRouter>
+        <FileList {...mockProps} />
+      </MemoryRouter>,
+    );
+    const sortButton = screen.getByLabelText('Sort by last modified');
+    const unsortedIcon = within(sortButton).getByLabelText('unsorted icon');
+    expect(unsortedIcon).toBeInTheDocument();
+  });
+
+  it('calls onSort with name for file name header', () => {
+    render(
+      <MemoryRouter>
+        <FileList {...mockProps} />
+      </MemoryRouter>,
+    );
+
+    const sortButton = screen.getByLabelText('Sort by name');
+    fireEvent.click(sortButton);
+
+    expect(mockProps.onSort).toHaveBeenCalledWith('name');
+  });
+
+  it('calls onDelete when a folder delete button is clicked', () => {
+    render(
+      <MemoryRouter>
+        <FileList {...mockProps} />
+      </MemoryRouter>,
+    );
+    fireEvent.click(screen.getAllByRole('button', { name: 'Delete' })[0]);
+    expect(mockProps.onDelete).toHaveBeenCalledWith({
+      ...mockFiles[0],
+      type: 'directory',
+    });
+  });
+
+  it('calls onDelete when a file delete button is clicked', () => {
+    render(
+      <MemoryRouter>
+        <FileList {...mockProps} />
+      </MemoryRouter>,
+    );
+    fireEvent.click(screen.getAllByRole('button', { name: 'Delete' })[1]);
+    expect(mockProps.onDelete).toHaveBeenCalledWith({ ...mockFiles[1] });
+  });
+
+  it('renders no rows when no files are present', () => {
+    render(
+      <MemoryRouter>
+        <FileList {...mockProps} data={[]} />
+      </MemoryRouter>,
+    );
+    expect(screen.getAllByRole('row').length).toBe(1);
+  });
+
+  it('copies file link to clipboard when copy button is clicked', async () => {
+    Object.assign(navigator, {
+      clipboard: {
+        writeText: jest.fn().mockResolvedValue(),
+      },
+    });
+
+    render(
+      <MemoryRouter>
+        <FileList {...mockProps} />
+      </MemoryRouter>,
+    );
+
+    const copyButton = screen.getAllByText('Copy link')[0];
+
+    fireEvent.click(copyButton);
+
+    expect(navigator.clipboard.writeText).toHaveBeenCalledWith(
+      'https://custom.domain.gov/~assets/report.pdf',
+    );
+
+    // Wait for the button text to change to "Copied!"
+    expect(await screen.findByText('Copied!')).toBeInTheDocument();
+    jest.useFakeTimers();
+    // Fast-forward 4 seconds and check that text reverts back
+    jest.advanceTimersByTime(4000);
+    expect(await screen.findByText('Copy link')).toBeInTheDocument();
```

Comment:
just realized this one will always pass because the other item still says copy link. need to update to check that _this_ item has reverted, or that the total number of "copy link" links is back to 2!